src/App.ts

```diff
@@ -6,84 +6,158 @@
+    await pagoPASoapClient.createPagamentiTelematiciPspNodoClient({
       endpoint: `${config.PAGOPA.HOST}:${config.PAGOPA.PORT}${
-        config.PAGOPA.SERVICES.PAYMENTS_CHECK
-      }`,
-      envelopeKey: PPTPortTypes.envelopeKey
+        config.PAGOPA.WS_SERVICES.PAGAMENTI
+      }?wsdl`
     })
   );
-  const attivaRPTPagoPaClient = new pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient(
-    await pagoPaSoapClient.createPagamentiTelematiciPspNodoClient({
-      endpoint: `${config.PAGOPA.HOST}:${config.PAGOPA.PORT}${
-        config.PAGOPA.SERVICES.PAYMENTS_ACTIVATION
-      }`,
-      envelopeKey: PPTPortTypes.envelopeKey
-    })
+
+  // Define a redis client necessary to handle persistent data
+  // for async payment activation process
+  const redisClient = redis.createClient(
+    config.REDIS_DB.PORT,
+    config.REDIS_DB.HOST
   );
 
-  // Define endpoints and start SOAP + RESTFUL WS
+  // Define RESTful and SOAP endpoints
   const app = express();
   app.set("port", config.CONTROLLER.PORT);
-  app.use(bodyParser.json());
-  app.use(bodyParser.urlencoded({ extended: false }));
-  setRestfulRoutes(app, config, verificaRPTPagoPaClient, attivaRPTPagoPaClient);
+  const loggerFormat =
+    ":date[iso] [info]: :method :url :status - :response-time ms";
+  app.use(morgan(loggerFormat));
+  setRestfulRoutes(app, config, redisClient, pagoPAClient);
+  getSoapServer(
+    redisClient,
+    config.PAYMENT_ACTIVATION_STATUS_TIMEOUT_SECONDS as number
+  )(app);
+
   const server = http.createServer(app);
   server.listen(config.CONTROLLER.PORT);
+
   logger.info(
     `Server started at ${config.CONTROLLER.HOST}:${config.CONTROLLER.PORT}`
   );
   return server;
 }
 
 export function stopServer(server: http.Server): void {
-  logger.info("Stopping Proxy PagoPa Server...");
+  logger.info("Stopping Proxy PagoPA Server...");
   server.close();
 }
 
-// Set Restful WS routes
+/**
+ * Set RESTful WS endpoints
+ * @param {core.Express} app - Express server to set
+ * @param {Configuration} config - PagoPa proxy configuration
+ * @param {redis.RedisClient} redisClient - The redis client used to store information sent by PagoPA
+ * @param {PagamentiTelematiciPspNodoAsyncClient} pagoPAClient - PagoPa SOAP client to call verificaRPT and attivaRPT services
+ */
 function setRestfulRoutes(
   app: core.Express,
   config: Configuration,
-  verificaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient,
-  attivaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient
+  redisClient: redis.RedisClient,
+  pagoPAClient: pagoPASoapClient.PagamentiTelematiciPspNodoAsyncClient
 ): void {
+  const jsonParser = bodyParser.json();
   app.get(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_CHECK,
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_REQUESTS_GET,
     (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Check Request (GET)...");
       toExpressHandler(
```

Comment:
by doing this, you will call `toExpressHandler` on *each* request - instead `toExpressHandler` should be called one to convert `PaymentController.getPaymentInfo(config.PAGOPA, pagoPAClient)` to an express handler and then pass the handler to `app.get()`, e.g.:

```ts
const getPaymentHandler = toExpressHandler(PaymentController.getPaymentInfo(config.PAGOPA, pagoPAClient));
app.get(
  config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_REQUESTS_GET,
  getPaymentHandler
);
```

this is true for all the handlers below.